### ISSUE TITLE
Cop name collision with rubocop-rspec

### ISSUE THREAD

**Issue by backus** [bbatsov/rubocop#3827]
I was playing with the current rubocop build and noticed a new warning:

```
/dev/projects/rubocop/.rubocop.yml: RSpec/FilePath has the wrong namespace - should be Rails
```

This might be confusing for users that depend on rubocop-rspec and rubocop itself. I think the proper solution here would be to not emit this warning if a cop exists for each case (`RuboCop::Cop::RSpec::FilePath` and `RuboCop::Cop::Rails::FilePath`). Before releasing the current build on master would you be open to fixing this? If you don't want this to be a release blocker it seems like it might also be valid to rename the rails cop to `Rails/RailsRoot` maybe.

**Comment by backus** [comment]
By the way I don't want to give the impression that I expect the rubocop maintainers to be aware of 3rd party dependencies like `rubocop-rspec`. This is basically just a wart brought on by #3414 and I figure it is easier for everyone if we catch this before issues are reported after a release

**Comment by Drenmi** [comment]
Thinking about this now can probably help inform our design decisions for the extension system, too. 😀

**Comment by bbatsov** [comment]
Yeah, I guess extensions should register their namespaces so their properly handled by RuboCop. Sadly we've made zero progress with #3414...

**Comment by Drenmi** [comment]
> Sadly we've made zero progress with #3414...

I've been doing some research. Still organizing my notes. I'm very busy at the moment though, so it's quite intermittent. 😓

**Comment by backus** [comment]
By the way, I think this bug can be resolved with some improvements to `CopStore`. I'm working on this now and I'll hopefully have a PR up soon

**Comment by backus** [comment]
This is actually still an issue. https://github.com/backus/rubocop-rspec/issues/299
